health/server_internal_test.go

```diff
@@ -81,3 +88,71 @@ func (s) TestShutdown(t *testing.T) {
 		t.Fatalf("status for %s is %v, want %v", testService, status, healthpb.HealthCheckResponse_NOT_SERVING)
 	}
 }
+
+// TestList verifies that List() returns the health status of all the services if no. of services are within
+// maxAllowedLimits
+func (s) TestList(t *testing.T) {
+	s := NewServer()
+	s.mu.Lock()
+	// Remove the zero value
+	delete(s.statusMap, "")
+	s.mu.Unlock()
+
+	// Fill out status map with information
+	const length = 3
+	for i := 1; i <= length; i++ {
+		s.SetServingStatus(fmt.Sprintf("%d", i),
+			healthpb.HealthCheckResponse_SERVING)
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+	var in healthpb.HealthListRequest
+	got, err := s.List(ctx, &in)
+
+	if err != nil {
+		t.Fatalf("s.List(ctx, &in) returned err %v, want nil", err)
+	}
+	if len(got.GetStatuses()) != length {
+		t.Fatalf("len(out.GetStatuses()) = %d, want %d",
+			len(got.GetStatuses()), length)
+	}
+	want := &healthpb.HealthListResponse{
+		Statuses: map[string]*healthpb.HealthCheckResponse{
+			"1": {Status: healthpb.HealthCheckResponse_SERVING},
+			"2": {Status: healthpb.HealthCheckResponse_SERVING},
+			"3": {Status: healthpb.HealthCheckResponse_SERVING},
```

Comment:
Let's make sure we add a test case for other states besides `SERVING` please.

Reply:
You got it!